common/djangoapps/third_party_auth/samlproviderconfig/views.py

```diff
@@ -0,0 +1,57 @@
+"""
+Viewset for auth/saml/v0/samlproviderconfig
+"""
+
+from edx_rbac.mixins import PermissionRequiredMixin
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from enterprise.models import EnterpriseCustomerIdentityProvider
+from rest_framework import permissions, viewsets
+from rest_framework.authentication import SessionAuthentication
+
+from openedx.features.enterprise_support.utils import fetch_enterprise_customer_by_id
+
+from ..models import SAMLProviderConfig
+from .serializers import SAMLProviderConfigSerializer
+
+
+class SAMLProviderMixin(object):
+    authentication_classes = [JwtAuthentication, SessionAuthentication]
+    permission_classes = [permissions.IsAuthenticated]
+    serializer_class = SAMLProviderConfigSerializer
+
+
+class SAMLProviderConfigViewSet(PermissionRequiredMixin, SAMLProviderMixin, viewsets.ModelViewSet):
+    """
+    A View to handle SAMLProviderConfig CRUD
+
+    Usage:
+        [HttpVerb] /auth/saml/v0/providerconfig/?enterprise-id=uuid
```

Comment:
It feels pretty strange to me to be requiring a query string for all HTTP requests. Typically the way I've seen most query strings used has been to filter on, and retrieve a subset of data strictly via GET calls. POSTs will typically have a body that gets written that I would expect to contain this information either as JSON or a form request. PATCH/PUT/DELETE I wouldn't expect this data to be necessary as they all require a primary key. 

I'm not positive what our access pattern is, but I would imagine that for any calls with an authorized user, we know exactly which enterprise_ids they're affiliated with, and we can just use those for returning the set of data that the user has access to instead of relying on user input. If we do want to keep this query string, I think that's fine for GET to just return the subset of provider configs that match the enterprise_customer_uuid and could see how that makes sense for a dashboard view page.

If we do decide to keep the query string required for all requests, I think it'll be helpful to have some 400 level error handling around if it's not specified as that seems like a helpful message to provide to downstream users of this API.